src/core_plugins/kibana/public/context/app.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
+
+import uiModules from 'ui/modules';
+import contextAppTemplate from './app.html';
+import './components/loading_button';
+import './components/size_picker/size_picker';
+import {
+  createInitialQueryParametersState,
+  QueryParameterActionsProvider,
+  QUERY_PARAMETER_KEYS,
+} from './query_parameters';
+import { QueryActionsProvider } from './query';
+
+const module = uiModules.get('apps/context', [
+  'elasticsearch',
+  'kibana',
+  'kibana/config',
+  'kibana/notify',
+  'ngRoute',
+]);
+
+module.directive('contextApp', function ContextApp() {
+  return {
+    bindToController: true,
+    controller: ContextAppController,
+    controllerAs: 'contextApp',
+    restrict: 'E',
+    scope: {
+      anchorUid: '=',
+      columns: '=',
+      indexPattern: '=',
+      predecessorCount: '=',
+      successorCount: '=',
+      sort: '=',
+      discoverUrl: '=',
+    },
+    template: contextAppTemplate,
+  };
+});
+
+function ContextAppController($scope, Private) {
+  const queryParameterActions = Private(QueryParameterActionsProvider);
+  const queryActions = Private(QueryActionsProvider);
+
+  this.state = createInitialState(this.discoverUrl);
+
+  this.actions = _.mapValues(Object.assign(
+    {},
+    queryParameterActions,
+    queryActions,
+  ), (action) => (...args) => action(this.state)(...args));
```

Comment:
I absolutely agree with splitting them up into files. That's exactly why I made the split between the query parameter actions and the query actions themselves.

I was referring to the repitition of the function names as well as the repeated call with `this.state`. The first is currently circumvented by just `Object.assign`ing everything to `actions` and the second is made DRY by the `_.mapValues`.

The fact that we have to assign them to `controller.actions` at all is just to make them available to the angular views where I can not just import them. 🙄 Would the following address your concerns?

```javascript
this.actions = _.mapValues(_.pick(queryActions, [
  'setAllRows',
  'fetchAllRowsWithNewQueryParameters',
  'fetchMorePredecessorRows',
  'fetchGivenPredecessorRows',
  'fetchMoreSuccessorRows',
  'fetchGivenSuccessorRows',
]), (action) => (...args) => action(this.state)(...args));
```